dask/array/core.py

```diff
@@ -4641,14 +4647,20 @@ def asanyarray(a, dtype=None, order=None, *, like=None, inline_array=False):
     >>> y = [[1, 2, 3], [4, 5, 6]]
     >>> da.asanyarray(y)
     dask.array<array, shape=(2, 3), dtype=int64, chunksize=(2, 3), chunktype=numpy.ndarray>
+
+    .. warning::
+        `dtype` is ignored if `a` has the attribute ``to_dask_array`` or
+        is a list or tuple of `Array`'s.
+        `order` is ignored if `a` is an `Array`, has the attribute ``to_dask_array``,
+        or is a list or tuple of `Array`'s.
     """
     if like is None:
         if isinstance(a, Array):
-            return a
+            return a if dtype is None else a.astype(dtype)
```

Comment:
Can you expand this into multiple lines? This breaks the pattern around it currently